instrumentation/nocode-testing/src/test/java/com/splunk/opentelemetry/instrumentation/nocode/YamlParserTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.instrumentation.nocode;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class YamlParserTest {
+  @Test
+  void testBasicRuleParsesOK() throws Exception {
+    String yaml = "- class: someClass\n" + "  method: someMethod\n";
+    assertEquals(1, YamlParser.parseFromString(yaml).size());
+  }
+
+  // formatting of the strings with newlines makes it easier to read and
+  // reason about the test.
+  // spotless:off
```

Comment:
an alternative that could be considered would be to use text blocks since the tests run with java 17

Reply:
I'll see what's available when I reformat this whole thing for `-contrib`.